rust/phoenix-sdk-core/src/sdk_client_core.rs

```diff
@@ -103,6 +105,42 @@ pub struct MarketMetadata {
     /// The adjustment factor is almost always 1, unless one base token is worth less than one quote atom (i.e. 1e-6 USDC)
     pub raw_base_units_per_base_unit: u32,
 }
+
+impl MarketMetadata {
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self> {
+        let (header_bytes, _) = bytes.split_at(size_of::<MarketHeader>());
+        let header = bytemuck::try_from_bytes::<MarketHeader>(header_bytes)
+            .map_err(|_| anyhow!("Failed to deserialize market header"))?;
+
+        let quote_lot_size = header.get_quote_lot_size().into();
+        let base_lot_size = header.get_base_lot_size().into();
+        let quote_multiplier = 10u64.pow(header.quote_params.decimals);
+        let base_multiplier = 10u64.pow(header.base_params.decimals);
+        let base_mint = header.base_params.mint_key;
+        let quote_mint = header.quote_params.mint_key;
+        let tick_size_in_quote_atoms_per_base_unit =
+            header.get_tick_size_in_quote_atoms_per_base_unit().into();
+        // max(1) is only relevant for old markets where the raw_base_units_per_base_unit was not set
+        let raw_base_units_per_base_unit = header.raw_base_units_per_base_unit.max(1);
+        let num_base_lots_per_base_unit =
+            base_multiplier * raw_base_units_per_base_unit as u64 / base_lot_size;
```

Comment:
@eugene-chen  Please check that this is correct.

(BA/rBU) * (rBU/BU) / (BA/BL) = BL/BU